docs/csharp/language-reference/compiler-messages/cs0106.md

```diff
@@ -36,11 +36,15 @@ The modifier 'modifier' is not valid for this item
  A class or interface member was marked with an invalid access modifier. The following examples describe some of these invalid modifiers:  
   
 -   The [static](../../../csharp/language-reference/keywords/static.md) and [public](../../../csharp/language-reference/keywords/public.md) modifiers are not permitted on interface methods.  
-  
+
+- The [static](../../../csharp/language-reference/keywords/static.md) modifier is not permitted on a local function. Whether a local function is static or instance is defined by its containing member.
+   
 -   The `public` keyword is not allowed on an explicit interface declaration. In this case, remove the `public` keyword from the explicit interface declaration.  
   
 -   The [abstract](../../../csharp/language-reference/keywords/abstract.md) keyword is not allowed on an explicit interface declaration because an explicit interface implementation can never be overridden.  
```

Comment:
Local functions also can't be `virtual`, `abstract`, `extern` or `partial`. Trying to do that does not currently cause CS0106, so this probably shouldn't be mentioned here. But I think it should do so, so I have filed https://github.com/dotnet/roslyn/issues/20279.